default-recommendations/for-loop-shortcuts.rkt

```diff
@@ -275,6 +275,17 @@
   ((~replacement loop-id.set-id #:original loop-id) clauses body ...))
 
 
+(define-refactoring-rule for/vector-with-in-range-to-length
+  #:description "Add `#:length` to `for/vector` loops to improve performance when the number of iterations is known."
+  #:literals (for/vector for*/vector in-range)
+  ((~or (~and for/vector (~bind [for-id #'for/vector]))
+        (~and for*/vector (~bind [for-id #'for*/vector])))
+   ([var:id (~or (in-range end:id)
+                 (in-range 0 end:id))])
```

Comment:
Instead of using `~or` here, you can use `~optional` to make this a little simpler. Also, use `~and` to capture the whole set of `for` clauses so you can preserve them unchanged in the output template.

```suggestion
   (~and for-clauses ([var:id (in-range (~optional 0) end:id)]))
```